example_test.go

```diff
@@ -335,6 +338,64 @@ func Example_jsonUnmarshalEmpty() {
 	// Uint64: false
 	// Uint: false
 	// Uintptr: false
+
+	t.Run("unmarshal null values to empty", func(t *testing.T) {
```

Comment:
Could you put this in a new example? Not a test please. The examples show up in docs and so help people understand how the library works.